lib/Target/AArch64/AArch64TargetMachine.cpp

```diff
@@ -495,7 +495,7 @@ bool AArch64PassConfig::addILPOpts() {
 }
 
 void AArch64PassConfig::addPreRegAlloc() {
-  if (TM->getOptLevel() != CodeGenOpt::None) { // TODO(yln): Maybe factor out the TM->getOptLevel() != CodeGenOpt::None check
+  if (TM->getOptLevel() != CodeGenOpt::None && TM->isPagerando()) {
```

Comment:
Push this to the end of `addPreRegAlloc` so it's even closer to register allocation?